test/BlockManager.js

```diff
@@ -644,5 +645,45 @@ describe('BlockManager', function () {
       const tx = blockManager.connect(signers[3]).claimBlockReward();
       assertRevert(tx, 'incorrect state');
     });
+    it('should not be able to finalise dispute if medians value is zero', async function () {
+      await mineToNextEpoch();
+      const votes = [0, 0, 0, 0, 0, 0, 0, 0, 0];
+      const epoch = await getEpoch();
+      const commitment = utils.solidityKeccak256(
+        ['uint32', 'uint48[]', 'bytes32'],
+        [epoch, votes, '0x727d5c9e6d18ed15ce7ac8d3cce6ec8a0e9c02481415c0823ea49d847ccb9ddd']
+      );
+      await voteManager.connect(signers[3]).commit(epoch, commitment);
+      await mineToNextState();// reveal
+      await voteManager.connect(signers[3]).reveal(epoch, votes,
+        '0x727d5c9e6d18ed15ce7ac8d3cce6ec8a0e9c02481415c0823ea49d847ccb9ddd');
+      await mineToNextState();// propose
+      const stakerIdAcc2 = await stakeManager.stakerIds(signers[3].address);
+      const staker = await stakeManager.getStaker(stakerIdAcc2);
+      const { biggestInfluencerId } = await getBiggestInfluenceAndId(stakeManager);
+      const iteration = await getIteration(voteManager, stakeManager, staker);
+      await blockManager.connect(signers[3]).propose(epoch,
+        [0, 0, 0, 0, 0, 0, 0, 0, 0],
+        iteration,
+        biggestInfluencerId);
+      await mineToNextState();// dispute
+      await blockManager.connect(signers[19]).giveSorted(epoch, 1, [7]);
+
+      const tx = blockManager.connect(signers[19]).finalizeDispute(epoch, 0);
+      assertRevert(tx, 'median can not be zero');
+    });
+    it('should be able to return correct data for getBlockMedians', async function () {
+      const tx = await blockManager.connect(signers[19]).getBlockMedians(await getEpoch());
+      assertDeepEqual(tx, [], 'transaction should return correct data');
+    });
+    it('getProposedBlock Function should work as expected', async function () {
+      const tx = await blockManager.connect(signers[19]).getProposedBlock(await getEpoch(), 0);
+      const medians = await blockManager.connect(signers[19]).getProposedBlockMedians(await getEpoch(), 0);
+      const { proposerId, iteration, biggestInfluence } = await blockManager.proposedBlocks(await getEpoch(), 0);
```

Comment:
Hey @adi44 
Here if I understand correctly 
We are comparing output from getProposedBlock() and blockManager.proposedBlocks().
As for both data sources is common only: contract. it would be true always. 
IMO better approach would be comparing on-chain data with the local one

Reply:
Yeah that can be done also, but here we are not checking if values are stored correctly. We want to check if block returns values as expected or not

Reply:
oh okay I see, cool @adi44 
Should we add an assertion to check if output of getProposedBlock is same as what you calculated locally?
With that whats stored will also be verified

Reply:
I think we have covered in earlier testcases

Reply:
where?

Reply:
Hey @adi44,
Is this solved?

Reply:
yes @abhishekvispute